go/interpreter/lfvm/instructions_test.go

```diff
@@ -1624,3 +1624,189 @@ func TestGeneralCall_ResultIsWrittenToStack(t *testing.T) {
 
 	}
 }
+
+func TestInstructions_InstructionsReturnErrorOnOverflow(t *testing.T) {
+	one := *uint256.NewInt(1)
+	zero := *uint256.NewInt(0)
+	maxUint256 := *uint256.NewInt(0).Sub(uint256.NewInt(0), uint256.NewInt(1))
+
+	tests := map[OpCode]struct {
+		implementation func(*context) error
+		inputs         []inputData
+	}{
+		JUMP: {
+			implementation: opJump,
+			inputs: []inputData{
+				inputStack(maxUint256),
+			},
+		},
+		JUMPI: {
+			implementation: opJumpi,
+			inputs: []inputData{
+				inputStack(one, maxUint256),
+			},
+		},
+		MSTORE: {
+			implementation: opMstore,
+			inputs: []inputData{
+				inputStack(zero, maxUint256),
+			},
+		},
+		MSTORE8: {
+			implementation: opMstore8,
+			inputs: []inputData{
+				inputStack(zero, maxUint256),
+			},
+		},
+		MCOPY: {
+			implementation: opMcopy,
+			inputs: []inputData{
+				inputStack(maxUint256, zero, zero),
+			},
+		},
+		MLOAD: {
+			implementation: opMload,
+			inputs: []inputData{
+				inputStack(maxUint256),
+			},
+		},
+		CALLDATACOPY: {
+			implementation: opCallDataCopy,
+			inputs: []inputData{
+				inputStack(maxUint256, zero, zero),
+				inputStack(one, one, maxUint256),
+			},
+		},
+		RETURNDATACOPY: {
+			implementation: opReturnDataCopy,
+			inputs: []inputData{
+				inputStack(maxUint256, zero, zero),
+				inputStack(zero, maxUint256, zero),
+				inputStack(one, zero, maxUint256),
+			},
+		},
+		CODECOPY: {
+			implementation: opCodeCopy,
+			inputs: []inputData{
+				inputStack(maxUint256, zero, zero),
```

Comment:
This only checks the `size` for an overflow, but not the two other parameters, which are offsets.

In the code, size overflows seem to be treated differently than offset overflows. Size overflows lead to `errOverflow` while other overflows seem to result in out-of-gas errors. I am wondering whether not all overflows related to offsets in memory accessed would lead to out-of-gas errors, and thus should not be explicitly checked.

tl;dr: are some of those extra checks for overflow errors that would actually also be called by out-of-gas errors?